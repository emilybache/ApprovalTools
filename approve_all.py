import os
import re
import shutil


def approve_all(directory):
    regex = re.compile("(.*\.)received(\..*)")
    for root, dirs, files in os.walk(directory):
        for filename in files:
            matches = re.findall(regex, filename)
            if matches:
                new_filename = matches[0][0] + "approved" + matches[0][1]
                shutil.copyfile(str(os.path.join(root, filename)),
                                str(os.path.join(root, new_filename)))


if __name__ == "__main__":
    import sys
    if len(sys.argv) > 1:
        directory = sys.argv[1]
    else:
        directory = os.getcwd()
    approve_all(directory)
