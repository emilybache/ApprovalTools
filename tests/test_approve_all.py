import os

from approvaltests import verify
from approve_all import *


def test_empty_dir(tmpdir):
    approve_all(tmpdir)
    verify(os.listdir(tmpdir))


def test_nothing_to_approve(tmpdir):
    approved_file = os.path.join(tmpdir, "a.approved.txt")
    with open(approved_file, "w") as f:
        f.write("foo")
    approve_all(tmpdir)
    verify(str(os.listdir(tmpdir)))


def test_approve_one_file(tmpdir):
    received_file = os.path.join(tmpdir, "a.received.txt")
    with open(received_file, "w") as f:
        f.write("foo")
    approve_all(tmpdir)

    all_files = os.listdir(tmpdir)
    verify(str(all_files))


def test_approve_one_file_with_distractions(tmpdir):
    received_file = os.path.join(tmpdir, "a.received.txt")
    with open(received_file, "w") as f:
        f.write("foo")
    other_file = os.path.join(tmpdir, "bar.xxx.txt")
    with open(other_file, "w") as f:
        f.write("bar")

    approve_all(tmpdir)

    all_files = os.listdir(tmpdir)
    verify(str(all_files))
