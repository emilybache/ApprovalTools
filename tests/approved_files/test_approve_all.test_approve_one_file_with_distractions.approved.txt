['a.received.txt', 'a.approved.txt', 'bar.xxx.txt']